File: leetcode3/염혜정/3168. Minimum Number of Chairs in a Waiting Room.java
class Solution {
    public int minimumChairs(String s) {
        int chair = 0;
        int max = 0;
        for (int i = 0; i<s.length(); i++) {
            if (s.charAt(i) == 'E') {
                chair++;
                max = Math.max(max, chair);
            }
            else if (s.charAt(i) == 'L') chair--;
        }
        return max;
    }
}

File: leetcode3/최원준/3168. Minimum Number of Chairs in a Waiting Room.py
#

'''
1. 아이디어 :
-

2. 시간복잡도 :
    O(n)

3. 자료구조/알고리즘 :
-

'''
class Solution:
    def minimumChairs(self, s: str) -> int:
        ans = 0
        curr = 0
        for char in s:
            if char == "E":
                curr += 1
                ans = max(ans, curr)
            else:
                curr -= 1
        return ans
